python/bigipconfigdriver.py

```diff
@@ -317,50 +314,24 @@ def _do_reset(self):
 
                 start_time = time.time()
 
-                config = _parse_config(self._config_file)
-                # No 'resources' indicates that the controller is not
-                # yet ready -- it does not mean to apply an empty config
-                if 'resources' not in config:
-                    continue
-
-                _handle_vxlan_config(config)
-                cfg_net = create_network_config(config)
                 incomplete = 0
-                for mgr in self._managers:
-                    partition = mgr.get_partition()
-                    cfg_ltm = create_ltm_config(partition, config)
-                    try:
-                        # Manually create custom profiles;
-                        # CCCL doesn't yet do this
-                        if 'customProfiles' in cfg_ltm and \
-                                mgr.get_schema_type() == 'ltm':
-                            tmp = 0
-                            customProfiles, tmp = _create_custom_profiles(
-                                mgr.mgmt_root(),
-                                partition,
-                                cfg_ltm['customProfiles'])
-                            incomplete += tmp
-
-                        # Apply the BIG-IP config after creating profiles
-                        # and before deleting profiles
-                        if mgr.get_schema_type() == 'net':
-                            incomplete += mgr._apply_net_config(cfg_net)
-                        else:
-                            incomplete += mgr._apply_ltm_config(cfg_ltm)
-
-                        # Manually delete custom profiles (if needed)
-                        if customProfiles and \
-                                mgr.get_schema_type() == 'ltm':
-                            _delete_unused_ssl_profiles(
-                                mgr.mgmt_root(),
-                                partition,
-                                cfg_ltm)
-
-                    except F5CcclError as e:
-                        # We created an invalid configuration, raise the
-                        # exception and fail
-                        log.error("CCCL Error: %s", e.msg)
-                        raise e
+                try:
+                    config = _parse_config(self._config_file)
+                    # No 'resources' indicates that the controller is not
+                    # yet ready -- it does not mean to apply an empty config
+                    if 'resources' not in config:
+                        continue
+                    incomplete = self._update_cccl(config)
+                except ValueError:
```

Comment:
Is this the only exception that can occur here? CCCL is supposed to catch all BIG-IP/SDK related exceptions, but if it misses something, we'll still have the same problem.